### PR TITLE
feat(cli): resume uploads across multiple `safe` runs

### PR DESCRIPTION
**Pull request by RolandSherwin**
### Summary generated by Reviewpad on 26 Oct 23 14:12 UTC
This pull request involves renaming the file "sn_cli/src/subcommands/files.rs" to "sn_cli/src/subcommands/files/mod.rs" and making changes to imports, module structure, function parameters, and method calls. Some code has been moved to a new module called "chunk_manager". The chunking functionality has been refactored and is now handled by the ChunkManager struct. The upload_files function has been modified to use the ChunkManager and no longer requires the wallet_dir_path parameter. The file_names_path has also been changed to use the root_dir. Some code related to logging has been commented out. Overall, the changes seem to improve the organization and modularity of the code.

The file `wallet.rs` has undergone several changes. Here is a summary of the changes made:
- Added an import statement for `ChunkManager` from the `files` module.
- Removed the import statement for `chunk_path` from the `files` module.
- Replaced occurrences of `wallet_dir_path` with `root_dir`.
- Replaced the `file_api` object initialization with `Files::new(client.clone(), root_dir.to_path_buf())`.
- Replaced usage of `chunk_path` function with `manager.chunk_path(&path)?`.
- Replaced getting all chunks from `chunked_files.values()` with `manager.get_chunks().iter()`.
- Removed the `ChunkedFile` struct definition and its associated code.

This file contains the implementation of a chunk manager, which is responsible for chunking files and managing the chunks. It includes several structs and methods for chunking files, storing chunk information, and retrieving chunks. The chunk manager takes a root directory and a files API as input and provides methods for chunking files, retrieving chunks, and marking finished chunks. The implementation uses external libraries such as color_eyre, rayon, sn_client, xor_name, walkdir for various functionalities.

Overall, this file provides the necessary functionality for chunking and managing files in the SAFE Network Software.

Please review these changes and consider their impact on the code. Let me know if you need more information.

### CODE REVIEW

File: sn_cli/src/subcommands/wallet.rs [hunk 1]
```diff
@@ -150,32 +147,26 @@ pub(crate) async fn wallet_cmds_without_client(cmds: &WalletCmds, root_dir: &Pat
 pub(crate) async fn wallet_cmds(
     cmds: WalletCmds,
     client: &Client,
-    wallet_dir_path: &Path,
+    root_dir: &Path,
```
Comment (jacderida) [inline sn_cli/src/subcommands/wallet.rs:150]: Hey Roland,

Minor point, but I remember changing this variable from `root_dir` -> `wallet_dir_path`. Why go back to something vague from something specific? The path does refer to the wallet directory does it not?

Cheers,

Chris

File: sn_protocol/src/test_utils.rs [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+//
+// This SAFE Network Software is licensed to you under The General Public License (GPL), version 3.
+// Unless required by applicable law or agreed to in writing, the SAFE Network Software distributed
+// under the GPL Licence is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied. Please review the Licences for the specific language governing
+// permissions and limitations relating to use of the SAFE Network Software.
+
+use std::fmt;
+
+pub fn assert_lists<I, J, K>(a: I, b: J)
```
Comment (joshuef) [inline sn_protocol/src/test_utils.rs]: can you elaborate or add docs here?

Comment (RolandSherwin) [reply sn_protocol/src/test_utils.rs]: Added docs to that function.

Comment (grumbach) [reply sn_protocol/src/test_utils.rs]: maybe `assert_list_eq` would be a better name?

File: sn_cli/src/subcommands/files/mod.rs [hunk 3]
```diff
@@ -224,36 +147,43 @@ async fn upload_files(
...
-    let temp_dir = tempdir()?;
-
-    // Payment shall always be verified.
-    let chunked_files = chunk_path(&file_api, &files_path, temp_dir.path()).await?;
-
-    let uploaded_file_info = chunked_files
-        .iter()
-        .map(|(file_addr, chunked_file)| (*file_addr, chunked_file.file_name.clone()))
-        .collect::<Vec<_>>();
+    // Return early if we hav no files to pay/verify
```
Comment (grumbach) [inline sn_cli/src/subcommands/files/mod.rs]:
```suggestion
    // Return early if we have no files to pay/verify
```

**Review by grumbach — APPROVED** [review]
LGTM

**Pull request merged**